dotnetv3/Cognito/Actions/CognitoWrapper.cs

```diff
@@ -0,0 +1,362 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier:  Apache-2.0
+
+using System.Net;
+
+namespace CognitoActions;
+
+/// <summary>
+/// Methods to perform Amazon Cognito Identity Provider actions.
+/// </summary>
+public class CognitoWrapper
+{
+    private readonly IAmazonCognitoIdentityProvider _cognitoService;
+
+    /// <summary>
+    /// Constructor for the wrapper class containing Amazon Cognito actions.
+    /// </summary>
+    /// <param name="cognitoService"></param>
+    public CognitoWrapper(IAmazonCognitoIdentityProvider cognitoService)
+    {
+        _cognitoService = cognitoService;
+    }
+
+    // snippet-start:[Cognito.dotnetv3.ListUserPools]
+    /// <summary>
+    /// List the Amazon Cognito user pools for an account.
+    /// </summary>
+    /// <returns>A list of UserPoolDescriptionType objects.</returns>
+    public async Task<List<UserPoolDescriptionType>> ListUserPoolsAsync()
+    {
+        var response = new ListUserPoolsResponse();
+        var request = new ListUserPoolsRequest { MaxResults = 10 };
+        var userPools = new List<UserPoolDescriptionType>();
+
+        do
+        {
+            response = await _cognitoService.ListUserPoolsAsync(request);
+
+            if (response.UserPools.Count > 0)
+            {
+                userPools.AddRange(response.UserPools);
+            }
+
+            request.NextToken = response.NextToken;
+        }
+        while (response.NextToken is not null);
+
+        return userPools;
+    }
+
+    // snippet-end:[Cognito.dotnetv3.ListUserPools]
+
+    // snippet-start:[Cognito.dotnetv3.ListUsers]
+    /// <summary>
+    /// Get a list of users for the Amazon Cognito user pool.
+    /// </summary>
+    /// <param name="userPoolId">The user pool Id.</param>
+    /// <returns>A list of users.</returns>
+    public async Task<List<UserType>> ListUsersAsync(string userPoolId)
+    {
+        var request = new ListUsersRequest
+        {
+            UserPoolId = userPoolId
+        };
+
+        var response = new ListUsersResponse();
+        var users = new List<UserType>();
+
+        do
+        {
+            response = await _cognitoService.ListUsersAsync(request);
+            users.AddRange(response.Users);
+            request.PaginationToken = response.PaginationToken;
+        }
+        while (response.PaginationToken is not null);
+
+        return users;
+    }
+
+    // snippet-end:[Cognito.dotnetv3.ListUsers]
+
+    // snippet-start:[Cognito.dotnetv3.AdminRespondToAuthChallenge]
+    public async Task<AuthenticationResultType> AdminRespondToAuthChallengeAsync(string userPoolId, string userName, string clientId, string mfaCode, string session)
```

Comment:
This method does not appear to be used, but in the spec it should be used at least once in the scenario. Could you re-check the spec and see if it is used in the correct place?

Reply:
Changed the call in the scenario to AdminRespondToAuthChallengeAsync from RespondToAuthChallengeAsync.